File: source_funcs/go_transcoder/CHECK_IF_X_CAN_GIVE_CHANGE_TO_EVERY_PERSON_IN_THE_QUEUE/CHECK_IF_X_CAN_GIVE_CHANGE_TO_EVERY_PERSON_IN_THE_QUEUE.go
package main

func f_gold(notes []int, n int) int {
	var (
		fiveCount int = 0
		tenCount  int = 0
	)
	for i := int(0); i < n; i++ {
		if notes[i] == 5 {
			fiveCount++
		} else if notes[i] == 10 {
			if fiveCount > 0 {
				fiveCount--
				tenCount++
			} else {
				return 0
			}
		} else {
			if fiveCount > 0 && tenCount > 0 {
				fiveCount--
				tenCount--
			} else if fiveCount >= 3 {
				fiveCount -= 3
			} else {
				return 0
			}
		}
	}
	return 1
}

File: rust_funcs/cpp_transcoder/CHECK_IF_X_CAN_GIVE_CHANGE_TO_EVERY_PERSON_IN_THE_QUEUE/CHECK_IF_X_CAN_GIVE_CHANGE_TO_EVERY_PERSON_IN_THE_QUEUE.rs
fn min(x: i32, y: i32) -> i32 {if x < y { x } else { y } }
fn max(x: i32, y: i32) -> i32 {if x > y { x } else { y }}
fn cmpfunc(a: &i32, b: &i32) -> std::cmp::Ordering {a.cmp(b)}
fn len(arr: &[i32]) -> usize {arr.len()}
fn sort(arr: &mut [i32]) {arr.sort_by(cmpfunc);}


fn f_gold(notes: &mut [i32], n: i32) -> i32 {
    let mut five_count = 0;
    let mut ten_count = 0;
    for i in 0..n {
        if notes[i as usize] == 5 {
            five_count += 1;
        } else if notes[i as usize] == 10 {
            if five_count > 0 {
                five_count -= 1;
                ten_count += 1;
            } else {
                return 0;
            }
        } else {
            if five_count > 0 && ten_count > 0 {
                five_count -= 1;
                ten_count -= 1;
            } else if five_count >= 3 {
                five_count -= 3;
            } else {
                return 0;
            }
        }
    }
    1
}
